src/core/authentication/authentication.service.ts

```diff
@@ -80,4 +104,79 @@ export class AuthenticationService {
 
     return agentInfo;
   }
+
+  public async extendSession(sessionToBeExtended: Session): Promise<boolean> {
+    let bearerToken: string;
+
+    try {
+      bearerToken = await getBearerToken(
+        this.kratosPublicUrl,
+        this.adminPasswordIdentifier,
+        this.adminPassword
+      );
+    } catch (e) {
+      const message = (e as Error)?.message;
+      this.logger?.error(
```

Comment:
I don't believe we should have this.logger.error anywhere here, we should be capturing an error, wrapping it in custom exception, and then rethrowing it. It's an exceptional flow, we should not reach that point (e.g. our account should be able to retrieve a bearer token)